core/src/main/java/io/undertow/server/handlers/resource/PathResource.java

```diff
@@ -283,7 +283,7 @@ public File getFile() {
 
     @Override
     public Path getFilePath() {
-        return file;
+        return file.normalize();
```

Comment:
Note this is for the failure on ubuntu, the returned path is in the form /./dir.txt and compared to /dir.txt, so the test fails. Unclear why this only happens on ubuntu though.